### PR TITLE
Add owner to luigi.task so we can send additional error emails

### PR DESCRIPTION
**Pull request by interskh**

### CODE REVIEW

**Comment by interskh** [comment]
This seems to address issue #12 too.

File: luigi/task.py [hunk 1]
```diff
@@ -128,6 +128,10 @@ class MyTask(luigi.Task):
     #: Defaults to 0 or value in luigi.cfg
     worker_timeout = None
 
+    #: Owner of the task: it is used for sending out error email in addition to
+    #: the one defined in luigi config
+    owner = None
```
Comment (Tarrasch) [inline luigi/task.py]: I would either have something like `owner_email = None` or let `owner` be some sort of class `luigi.TaskOwner` with fields like `email` and such. Then organizations can add whatever fields they want to their `TaskOwner` subclass class.

**Comment by Tarrasch** [comment]
Thanks for this initative! I really like it, just see my comment and let me know what you think. :)

**Comment by mvj3** [comment]
:+1:

File: luigi/task.py (hunk 1 above)
Comment (interskh) [inline luigi/task.py]: sounds reasonable, i will just change it to owner_email to keep simple :)

**Comment by interskh** [comment]
seems like some tests failed for python 3.3.. i will take a look :(

**Comment by interskh** [comment]
Finally it passed the tests.. Let me know if that looks okay for you now @Tarrasch

File: test/email_test.py [hunk 2]
```diff
@@ -20,6 +20,8 @@
 from helpers import with_config
 from luigi import notifications
 
+import six
```
Comment (Tarrasch) [inline test/email_test.py]: `from luigi import six`

File: luigi/task.py [hunk 3]
```diff
@@ -128,6 +128,10 @@ class MyTask(luigi.Task):
     #: Defaults to 0 or value in luigi.cfg
     worker_timeout = None
 
+    #: Owner of the task: it is used for sending out error email in addition to
+    #: the one defined in luigi config
+    owner_email = None
```
Comment (Tarrasch) [inline luigi/task.py]: Can we make this a `@property` with actual sphinx docs?

File: luigi/notifications.py [hunk 4]
```diff
@@ -157,22 +157,34 @@ def send_email(subject, message, sender, recipients, image_png=None):
...
+        logger.info("Sending warning email to %r", recipients)
         send_email(
             subject=subject,
             message=message,
             sender=sender,
-            recipients=(receiver,)
+            recipients=recipients
         )
     else:
         logger.info("Skipping error email. Set `error-email` in the `core` "
```
Comment (ulzha) [inline luigi/notifications.py]: Update this hint to also include the new way?

**Comment by ulzha** [comment]
I wish there were a convention or something to distinguish opaque "data management/monitoring" properties (like email, SLA, ...) from those that are technically related to task business logic, or scheduling/performance optimization logic. Haven't come up with a brillant vision for that yet, so +1.

**Comment by Tarrasch** [comment]
@interskh, mind doing the last iteration so this can be merged? :)

**Comment by interskh** [comment]
@Tarrasch sorry was a bit busy recently. Yah, will make the changes :)

**Comment by interskh** [comment]
@Tarrasch made the change! The tests failed but doesn't seem like it was my change.

**Comment by Tarrasch** [comment]
All python 3.x runs are failing. I'm sure it's no coincidence?

**Comment by interskh** [comment]
Yes but it is failing at pip install 😑

**Comment by Tarrasch** [comment]
Oh I see. I couldn't check the logs since my internet is poor at the moment. Hmm. It has to be fixed then. Not sure when I'll have time for it though :-/

**Comment by interskh** [comment]
Failed on building sasl:
```
  gcc -pthread -Wno-unused-result -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -fPIC -Isasl -I/opt/python/3.3.5/include/python3.3m -c sasl/saslwrapper.cpp -o build/temp.linux-x86_64-3.3/sasl/saslwrapper.o

  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]

  sasl/saslwrapper.cpp: In member function ‘void saslwrapper::ClientImpl::interact(sasl_interact_t*)’:

  sasl/saslwrapper.cpp:341:11: warning: unused variable ‘input’ [-Wunused-variable]

  gcc -pthread -Wno-unused-result -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -fPIC -Isasl -I/opt/python/3.3.5/include/python3.3m -c sasl/saslwrapper_wrap.cxx -o build/temp.linux-x86_64-3.3/sasl/saslwrapper_wrap.o

  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]

  sasl/saslwrapper_wrap.cxx: In function ‘void SWIG_Python_AddErrorMsg(const char*)’:

  sasl/saslwrapper_wrap.cxx:884:58: error: ‘PyString_AsString’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx: In function ‘PySwigClientData* PySwigClientData_New(PyObject*)’:

  sasl/saslwrapper_wrap.cxx:1275:26: error: ‘PyClass_Check’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx: In function ‘PyObject* PySwigObject_format(const char*, PySwigObject*)’:

  sasl/saslwrapper_wrap.cxx:1348:47: error: ‘PyString_FromString’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx:1350:33: error: ‘PyString_Format’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx: In function ‘PyObject* PySwigObject_repr(PySwigObject*)’:

  sasl/saslwrapper_wrap.cxx:1380:105: error: ‘PyString_AsString’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx:1380:106: error: ‘PyString_FromFormat’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx:1388:37: error: ‘PyString_ConcatAndDel’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx: In function ‘int PySwigObject_print(PySwigObject*, FILE*, int)’:

  sasl/saslwrapper_wrap.cxx:1402:33: error: ‘PyString_AsString’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx: In function ‘PyObject* PySwigObject_str(PySwigObject*)’:

  sasl/saslwrapper_wrap.cxx:1415:31: error: ‘PyString_FromString’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx: In function ‘PyTypeObject* _PySwigObject_type()’:

  sasl/saslwrapper_wrap.cxx:1624:6: error: ‘coercion’ was not declared in this scope

  sasl/saslwrapper_wrap.cxx:1624:15: error: expected ‘}’ before numeric constant

  sasl/saslwrapper_wrap.cxx:1624:15: error: expected ‘,’ or ‘;’ before numeric constant

  sasl/saslwrapper_wrap.cxx:1604:15: warning: unused variable ‘swigobject_doc’ [-Wunused-variable]

  sasl/saslwrapper_wrap.cxx:1606:26: warning: unused variable ‘PySwigObject_as_number’ [-Wunused-variable]

  sasl/saslwrapper_wrap.cxx:1637:3: warning: no return statement in function returning non-void [-Wreturn-type]

  sasl/saslwrapper_wrap.cxx: At global scope:

  sasl/saslwrapper_wrap.cxx:1641:3: error: expected unqualified-id before ‘if’

  In file included from /usr/include/c++/4.6/stdexcept:39:0,

                   from sasl/saslwrapper_wrap.cxx:2542:

  /usr/include/c++/4.6/exception:37:37: error: expected ‘}’ before end of line

  /usr/include/c++/4.6/exception:37:37: error: expected declaration before end of line

  error: command 'gcc' failed with exit status 1

  

  ----------------------------------------

  Failed building wheel for sasl
```

**Comment by Tarrasch** [comment]
Do you want to investigate this @interskh ?

**Comment by interskh** [comment]
Looks like other ppl are seeing the same issue with python3 (https://github.com/BradRuderman/pyhs2/issues/44 http://stackoverflow.com/questions/31701493/problems-installing-pyhs2-in-python3), no solution has been found. Could be some error on sasl package.

**Comment by Tarrasch** [comment]
Hmm, what depedency just changed then? Did Travis upgrade to newer Ubuntu or was it a too slack pip depedency in luigi?

**Comment by interskh** [comment]
Looks like sasl is introduced to snakebite in 2.6.0+ (released yesterday).. Also snakebite not supporting python 3 so not sure what is the best way to do this. Is there a way to skip all snakebite stuff for python3?

**Comment by Tarrasch** [comment]
This looks really nice. Can you please just rebase so the build works?

**Comment by interskh** [comment]
Just squashed and rebased :smile:

**Comment by Tarrasch** [comment]
Nice! :)

**Pull request merged**